internal/services/storage/storage_account_resource.go

```diff
@@ -691,6 +691,7 @@ func resourceStorageAccount() *pluginsdk.Resource {
 						"smb": {
 							Type:     pluginsdk.TypeList,
 							Optional: true,
+							Computed: true,
```

Comment:
This shouldn't be marked as `computed`, if all the inner members are empty/have a default value of `false` (in the case of `multichannel_enabled`) then we should be returning an empty interface when flattening the block - so can we update this here instead?

We shouldn't really be marking any new fields as Optional & Computed at this point, since we're going to need to undo that in the future (instead, using a Delta/Patch Update method, users should specify `ignore_changes` if they wish to ignore the value for a field - however we need to flatten these correctly, per-above).

As such would you mind updating this?

Reply:
If I understand correctly, you meant to add something similar to:

```
+       // The service API returns smb for premium file storage account no matter it is specified in the creation request.
+       // The returned default value in this case only has the `smb.multichannel.enabled = false`.
+       // We need to explicitly check for above case to ensure we didn't write this "unset" block back to the state file, which wil result into a plan diff.
+       if input.Versions == nil && input.ChannelEncryption == nil && input.AuthenticationMethods == nil && input.KerberosTicketEncryption == nil {
+               if input.Multichannel == nil {
+                       return []interface{}{}
+               }
+               if input.Multichannel.Enabled == nil || !*input.Multichannel.Enabled {
+                       return []interface{}{}
+               }
+       }
```

at the head of the `flattenedSharePropertiesSMB`.

Though that can solve the reported issue when users didn't specify the `smb` block, whilst it will break the other half of users who have explicitly specified the `smb.multichannel_enabled = false` as below:

```
  share_properties {
    smb {
      multichannel_enabled = false
    }
    retention_policy {
      days = 5
    }
  }
```

The terraform plan then complains:

```shell
      ~ share_properties {

          + smb {
              + multichannel_enabled = false
            }

            # (1 unchanged block hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```